keanu-project/src/main/java/io/improbable/keanu/vertices/utility/PrintVertex.java

```diff
@@ -0,0 +1,42 @@
+package io.improbable.keanu.vertices.utility;
+
+import com.google.common.base.Preconditions;
+import io.improbable.keanu.annotation.ExportVertexToPythonBindings;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.NonProbabilistic;
+import io.improbable.keanu.vertices.NonSaveableVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import java.io.PrintStream;
+
+public class PrintVertex extends DoubleVertex implements NonProbabilistic<DoubleTensor>, NonSaveableVertex {
```

Comment:
I think this can be a generic vertex. You don’t care about the data type of your parent. This also means that the .print() method might make sense to be on the base vertex class, which is interesting and we might need to have a think/chat about that.

Reply:
made generic as discussed offline